device/transport/mqtt/src/mqtt.ts

```diff
@@ -657,7 +657,7 @@ export class Mqtt extends EventEmitter implements DeviceTransport {
     let baseConfig: MqttBaseTransportConfig = {
       uri: 'mqtts://' + (credentials.gatewayHostName || credentials.host),
       username: credentials.host + '/' + clientId +
-        '/' + endpoint.versionQueryString().substr(1) +
+        '/' + endpoint.versionQueryString() +
```

Comment:
>endpoint.versionQueryString() [](start = 14, length = 29)

I'll approve it, but I guess I'm wondering why you went through the effort of removing it originally.